cloudsql/mysql/database-sql/cloudsql.go

```diff
@@ -0,0 +1,264 @@
+	"html/template"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"runtime"
+	"strconv"
+
+	"github.com/go-sql-driver/mysql"
+)
+
+// db is global database connection, parsedTemplate is parsed HTML template.
+var db *sql.DB
+var parsedTemplate *template.Template
+
+// The vote struct stores a row from the votes table in the Cloud SQL instance.
+// Each vote includes a candidate ("TABS" or "SPACES") and a timestamp.
+type vote struct {
+	Candidate string
+	VoteTime  mysql.NullTime
+}
+
+// The templateData struct is used to pass data to the HTML template.
+type templateData struct {
+	TabsCount   uint
+	SpacesCount uint
+	VoteMargin  string
+	RecentVotes []vote
+}
+
+func main() {
+	var err error
+
+	parsedTemplate, err = template.ParseFiles("templates/index.html")
+	if err != nil {
+		log.Fatalf("unable to parse template file: %s", err)
+	}
+
+	db, err = initConnectionPool()
+	if err != nil {
+		log.Fatalf("initConnectionPool: unable to initialize database connection pool: %s", err)
+	}
+
+	if err = initDBSchema(); err != nil {
+		log.Fatalf("unable to create table: %s", err)
+	}
+
+	http.HandleFunc("/", indexHandler)
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+
+	log.Printf("Listening on port %s", port)
+	if err := http.ListenAndServe(":"+port, nil); err != nil {
+		log.Fatal(err)
+	}
+
+}
+
+// indexHandler handles requests to the / route.
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		err := showTotals(w, r)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(err.Error()))
+		}
+	case "POST":
+		saveVote(w, r)
+	default:
+		http.Error(w, fmt.Sprintf("HTTP Method %s Not Allowed", r.Method), http.StatusMethodNotAllowed)
+	}
+}
+
+// recentVotes returns a slice of the last 5 votes cast.
+func recentVotes() ([]vote, error) {
+	var votes []vote
+	rows, err := db.Query(`SELECT candidate, time_cast FROM votes ORDER BY time_cast DESC LIMIT 5`)
+	if err != nil {
+		return votes, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		nextVote := vote{}
+		rowError := rows.Scan(&nextVote.Candidate, &nextVote.VoteTime)
+		if err != nil {
+			return votes, rowError
+		}
+		votes = append(votes, nextVote)
+	}
+	if err = rows.Err(); err != nil {
+		return votes, err
+	}
+	return votes, nil
+}
+
+// currentTotals returns a templateData structure for populating the web page.
+func currentTotals() (templateData, error) {
+
+	// get total votes for each candidate
+	var tabVotes, spaceVotes uint
+	err := db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='TABS'`).Scan(&tabVotes)
+	if err != nil {
+		return templateData{}, err
+	}
+	err = db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='SPACES'`).Scan(&spaceVotes)
+	if err != nil {
+		return templateData{}, err
+	}
+
+	// voteMargin is string representation of the current voting margin,
+	// such as "1 vote" (singular) or "2 votes" (plural).
```

Comment:
Given margin is a synonym of diff, I might name this `voteDiffLabel` for more clarity.

Another approach in keeping with golang short variable names is to move this logic to a `voteDiff` type: https://play.golang.org/p/1UA016ywIEd

Reply:
Done - created a voteDiff type per your sample. (Thanks!)